### ISSUE TITLE
Notes/links  to get started?

### ISSUE THREAD

**Issue by hth2** [fucm/node-red#1]
can you please add some brief notes or links on how to use this? I am very interested in using this module to read data from my heat pump, but I am not that experienced with Node-Red and don't know to get started. I have managed to configure my HA instance with a bunch of sensors.

**Comment by fucm** [comment]
Hi @hth2, I have added a readme file to the repo, can you check if this is already enough information for you? Let me know, which step fails for you.

**Comment by hth2** [comment]
Hi @fucm, many thanks for adding the notes in readme, I find it very helpful.

I tried to get as long as I could:

 > 2. Setup the MQTT Brocker
>   i. Add MQTT integration to HA
>   ii. Make sure HA is able to receive MQTT messages

I followed the official doc for Mosquitto MQTT add-on for HA, didn't get any error/problem but I don't know how to verify it's working

> 3. Setup Node-RED
>   i. Import the two flows in this repo into Node-RED
> ii. Without InfluxDB: delete the nodes which prepare and send the data to InfluxDB. You don't need them.

I disabled the nodes, just in case I decide to add influx later

> iii. Update the node: "HA MQTT Set device details" with your installation details.
> iv. Verify or add the MQTT broker configuration

again I don't know how to do this step

> v. Deploy the flows

I did it, and see errors like these:

> "Error: getaddrinfo ENOTFOUND servicewelt.netzwelt"

I changed servicewelt.netzwelt to the actual IP of the ISG. I also added user authentication, since I could not figure out how to disable user-auth for ISG

> 4. Check HA for new sensors

didn't find any; I guess the MQTT broker is not working yet

thanks again for your help.

**Comment by fucm** [comment]
Hi @hth2 ,

In HA, if you did setup MQTT, you can go to "Configuration > Integrations > MQTT > CONFIGURE" and start listening to the topic "homeassistant/sensor/#". You should see a bunch of messages, which start with something like "homeassistant/sensor/...".

If you don't see them, I would go to node-red again. The debug panel will only show something, if an error occurs or if you did enable debug msg nodes. You can do so in the "HA MQTT Discovery" flow. Then redeploy it and you should soon get some messages showing up in the node-red debug log. 

In case you see the messages, but still do not receive anything in HA, it's probably your MQTT broker config in node-red, which is not working. 

In case you don't see any messages, copy the debug node and attach it anywhere earlier in the flow to find if somewhere the messages are flowing.

**Comment by azrael783** [comment]
I implemented the flows also and I did recognize that the http get node is pointing to http://servicewelt.netzwelt/?s=1,1 which I guess is due to your network setup. I had to change it to http://servicewelt/?s=1,1 to get the data fetched from the ISG

**Comment by hth2** [comment]
Hi @fucm, thanks for your guidance.

I got to the point where I can see the msgs when I click on "Start listening" in "Configuration > Integrations > MQTT > CONFIGURE". 

How can I make use of those msgs as sensors for HA? Should I manually add to configuration.yaml something like

```
sensor:
  - platform: mqtt
    state_topic: "homeassistant/sensor/heatpump/raumtemperatur"
```
for each value I want to monitor?

**Comment by azrael783** [comment]
@hth2 Normally HA should autodiscover them. But this seems not to work (see second issue I opened).

**Comment by fucm** [comment]
@hth2 You do not need to add anything to a config file, if the MQTT integration is added to HA and if it is working. I did update the flow with a trigger node, which immediately flushes all discovery messages. There is a delay node, which keeps similar messages back for one hour to avoid to much traffic. 

And as @azrael783 identified correctly, the URL of the ISGWeb needs to be updated as well. I did update the flow and the README file accordingly.

When you see the messages in HA ending with /config it actually should work and add all the sensor automatically to your default lovelance dashboard.

**Comment by hth2** [comment]
finally I can see the values as sensors in HA.

Shall I commit my flows and make a pull request? I think it can be helpful for people with the same device as I have.

**Comment by fucm** [comment]
Hi @hth2, just give it a try. It depends on your proposed changes, if I can just pull in the request.

What was finally the issue?

**Comment by hth2** [comment]
> Hi @hth2, just give it a try. It depends on your proposed changes, if I can just pull in the request.
> 
> What was finally the issue?

I am not sure to be honest; it seems the msgs sent to MQTT were not properly prepared. When I edited the nodes to match the data from my ISG Web, things started to work.

**Comment by tomasvida** [comment]
Hello,
I installed ServiceWelt Tecalor flow, configured my ISG and I can't get through this error in the CONVERT To NUMBER cell with this error message:
14. 1. 2022 21:58:19node: Convert to number
msg : string[73]
"Invalid JSONata expression: Unable to cast value to a number: "87.496kWh""
I also tried the variant where I exchanged the dot for a comma in the number, but it ends with the same error voice.
Can you help me, please?
Tnx. Tomas

**Comment by comewhatmay** [comment]
Hi @tomasvida, I have had the same problems. I seems to me that ISG web site was changed and there are now no spaces between the number and the unit. 
The old flow uses the space to extract the number substring ($substringBefore(payload[0], ' ')) in the "Convert to number" change function.
I have changed this to $substringBefore(payload[0], '°') for values with °C and additional rules for the other units:
![image](https://user-images.githubusercontent.com/24809726/152692715-6ddfa56b-6f31-48b4-a981-d4f8fdb828ab.png)

There may be better ways to do this, e.g. use regular expressions, but this works for me.
